### PR TITLE
Custom chart upload

### PR DESCRIPTION
**Pull request by vivek-devtron**
# Description
Custom chart upload functionality so that user can upload there custom chart in the system and configure app through that chart

Fixes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B
- [x] https://docs.google.com/spreadsheets/d/1QU9pr9OQRHaJ41Hvl07Pl3Ak25nQvkj-1aAM8-6PZNc/edit#gid=0

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [x] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/CustomChart/types.tsx [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import { ResponseType } from '../../services/service.types'
+export interface chartDetailType {
+    chartDescription: string
+    name: string
+    version: string
+    count?: number
+}
+export interface chartUploadType {
```
Comment (sohel-rp) [inline src/components/CustomChart/types.tsx:8]: same here

File: src/components/CustomChart/types.tsx [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import { ResponseType } from '../../services/service.types'
+export interface chartDetailType {
```
[the remark this answers is not in the document]
Comment (sohel-rp) [reply src/components/CustomChart/types.tsx:2]: And file should be of `ts` format/extension.

File: src/components/CustomChart/customChart.service.tsx [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+import { ChartListResponse, ChartUploadResponse } from './types'
+
+export const getChartList = (): Promise<ChartListResponse> => {
+    return get(Routes.CUSTOM_CHART_LIST)
+}
+
+export const validateChart = (payload): Promise<ChartUploadResponse> => {
+    return post(Routes.VALIDATE_CUSTOM_CHART, payload, {}, true)
+}
+export const uploadChart = (payload): Promise<ChartUploadResponse> => {
```
Comment (sohel-rp) [inline src/components/CustomChart/customChart.service.tsx:12]: Please provide the request/payload type.

File: src/components/CustomChart/customChart.service.tsx [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import { Host, Routes } from '../../config'
+import { post, get, put } from '../../services/api'
+import { ChartListResponse, ChartUploadResponse } from './types'
+
+export const getChartList = (): Promise<ChartListResponse> => {
+    return get(Routes.CUSTOM_CHART_LIST)
+}
+
+export const validateChart = (payload): Promise<ChartUploadResponse> => {
```
Comment (sohel-rp) [inline src/components/CustomChart/customChart.service.tsx:9]: same here

File: src/components/globalConfigurations/GlobalConfiguration.tsx [hunk 5]
```diff
@@ -276,38 +285,44 @@ function Body({ getHostURLConfig, checkList, serverMode, handleChecklistUpdate }
...
                 path={URLS.GLOBAL_CONFIG_CHART}
                 render={(props) => {
-                    return <ChartRepo />;
+                    return <ChartRepo />
+                }}
+            />
+            <Route
+                path={URLS.GLOBAL_CONFIG_CUSTOM_CHARTS}
+                render={(props) => {
+                    return <CustomChartList />
```
Comment (sohel-rp) [inline src/components/globalConfigurations/GlobalConfiguration.tsx:301]: Use the other syntax for the route. I.e.
```
<Route path={URLS.GLOBAL_CONFIG_CUSTOM_CHARTS}>
     <CustomChartList />
</Route>
```

File: src/css/base.scss [hunk 6]
```diff
@@ -1661,6 +1665,10 @@ button.anchor {
     border-bottom: solid 1px var(--N200);
 }
 
+.border-bottom-n1 {
```
Comment (shivani170) [inline src/css/base.scss:1668]: we already have this class as `border-bottom`

File: src/components/EmptyState/emptyState.scss [hunk 7]
```diff
@@ -72,4 +76,4 @@
     // .subtitle a:hover {
```
Comment (shivani170) [inline src/components/EmptyState/emptyState.scss:76]: Remove this uncommented part

Comment (vivek-devtron) [reply src/components/EmptyState/emptyState.scss:76]: This old code. someone marked it as todo. that's why left it there

File: src/css/base.scss (hunk 6 above)
Comment (vivek-devtron) [reply src/css/base.scss:1668]: it is with different color

**Review by sohel-rp — APPROVED** [review]

**Pull request merged**